pay-api/src/pay_api/services/eft_transactions.py

```diff
@@ -36,36 +42,80 @@ class EFTTransactions:
     """Service to manage EFT Transactions."""
 
     @staticmethod
-    def get_remaining_credits(short_name_id: int):
-        """Return the remaining credit for a short name."""
-        return db.session.query(func.sum(EFTCreditModel.remaining_amount))\
-            .filter(EFTCreditModel.short_name_id == short_name_id)\
-            .group_by(EFTCreditModel.short_name_id).scalar()
+    def get_account_name():
+        """Return case statement for deriving payment account name."""
+        return case(
+            (PaymentAccountModel.name.like('%-' + PaymentAccountModel.branch_name),
+             func.replace(PaymentAccountModel.name, '-' + PaymentAccountModel.branch_name, '')
+             ), else_=PaymentAccountModel.name).label('account_name')
+
+    @staticmethod
+    def get_funds_received_query():
+        """Return the EFT transaction funds received."""
+        return db.session.query(EFTTransactionModel.id.label('transaction_id'),
```

Comment:
What's the intent of the null columns?

Reply:
Good call, I needed this to get a proper union for Statement Paid vs Funds Received. 

I will add some comments to clarify.

Reply:
all good very minor

Reply:
Added some comments above